### ISSUE TITLE
[v3] Whisparr doesn't get results from 1337x via Jackett torznab feeb

### ISSUE THREAD

**Issue by fangly-fish** [Whisparr/Whisparr#275]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

I have configured 1337x.to in my jackett instance, and when I search it in jackett I see lots of results. I have also pointed my the jackett search plugin in qbittorrent and am able to see results from 1337x through that search plugin. But configuring the 1337x torznab feed from jackett as a torznab indexer in Whisparr v3 fails with an error that no results were returned. Perhaps the test query that Whisparr is sending needs to be tweaked a little?

### Expected Behavior

The test query from Whisparr to the torznab feed for 1337x from jackett should include results and allow me to save the indexer.

### Steps To Reproduce

1. Setup a Jackett instance and configure 1337x tracker
2. In Whisparr v3 add a torznab indexer configured with the Jackett torznab URL and API key
3. Click test button

### Environment

```markdown
- OS: Synology DSM 7.2
- Whisparr: 3.0.0.628
- Docker Install: hotio/whisparr:v3
- Using Reverse Proxy: yes
- Browser: chrome on windows
- Database: Sqlite 3.44.2
```

### What branch are you running?

Other (This issue will be closed)

### Trace Logs?

I set the log level per the wiki, cleared the logs, and tried to add the indexer. This is all that was put in the log file (note here that the log file was still named whisparr.txt, not whisparr.trace.txt:

```
2024-10-23 19:22:05.3|Warn|WhisparrErrorPipeline|Invalid request Validation failed: 
 -- : Query successful, but no results in the configured categories were returned from your indexer. This may be an issue with the indexer or your indexer category settings.
```

### Anything else?

I know the branch field says the issue will be closed, but I also asked about this on Discord support channel with no response, and I've seen other people asking about the same in Discord.

Manual search of the tracker in Jackett

![Screenshot 2024-10-23 192401](https://github.com/user-attachments/assets/cf67d0c4-6aff-44d8-b475-8f747f6ae53d)

Search results  from Jackett, filtered on 1337x, in qBittorrent

![Screenshot 2024-10-23 192512](https://github.com/user-attachments/assets/c3a07afd-3e26-4b79-99c0-7f9c5fe1e44a)

The Whisparr error

![Screenshot 2024-10-23 192422](https://github.com/user-attachments/assets/ab5c8404-6875-4b27-8044-b30a00be6b3f)

The Jackett log for the manual search and the Whisparr test querry

![Screenshot 2024-10-23 193324](https://github.com/user-attachments/assets/ff8a8bcd-57b4-4229-97de-8bcf4d7c1b3f)

**Comment by sampulsar** [comment]
@fangly-fish 

Try no search term as that is how the arr applications test the indexes. When you do this 133x returns no results. 

The blank search term is used as it works for most of them and if there was a different term it may not work for more indexers.

This issue would have been already covered by the other arr applications, as you can click on save twice if the test returns a false negative.

If you really need this functionality, to enter a test term,  then it should be request it within Radarr, so that the change can be fixed upstream. This means it is done once.

**Comment by fangly-fish** [comment]
> as you can click on save twice if the test returns a false negative.

Ah, I didn't know about that, don't see it mentioned [here](https://wiki.servarr.com/whisparr/settings#indexers). Doesn't appear to be working for me, though. I've tried doubel-clicking without waiting for the save to fail, and waiting for the save to fail before the second click, and clicking like 5 times with varying waits. Should be that be filed upstream?

**Comment by sampulsar** [comment]
@fangly-fish

The double-click maybe is the wrong wording, as you click on the save and wait for the validation to fail, and then you can click on save again without changing any values and it will add the indexer.

**Comment by fangly-fish** [comment]
Just updated to 3.0.0.648. Waiting for the save button to validate and then flash the red exclamation before clicking the button again still doesn't save the indexer.

**Comment by GasBandit** [comment]
Also not working for kickasstorrents.ws - same issue. "Query successful, but no results in the configured categories were returned from your indexer. "  Indexer definitely returns results in the categories when searched in Jackett itself.

**Comment by plz12345** [comment]
Provide trace logs reproducing the empty search, if you want this to be looked at further.  Not everyone has access the indexers you do, nor uses Jackett.

**Comment by GasBandit** [comment]
> Provide trace logs reproducing the empty search, if you want this to be looked at further. Not everyone has access the indexers you do, nor uses Jackett.

Fair enough - here's a couple trace logs - the first one is for the kickasstorrents.ws torznab in Jackett that doesn't work
2025-03-28 21:45:24.9|Trace|Http|Req: 3977 [POST] /api/v3/indexer/test (from 195.*.*.48 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:136.0) Gecko/20100101 Firefox/136.0)
2025-03-28 21:45:24.9|Debug|Torznab|Downloading Feed https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/kickasstorrents-ws/results/torznab/api?t=search&cat=6000,103510&extended=1&apikey=(removed)&offset=0&limit=100
2025-03-28 21:45:24.9|Trace|HttpClient|Req: [GET] https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/kickasstorrents-ws/results/torznab/api?t=search&cat=6000,103510&extended=1&apikey=(removed)&offset=0&limit=100
2025-03-28 21:45:24.9|Trace|ConfigService|Using default config value for 'proxyenabled' defaultValue:'False'
2025-03-28 21:45:25.1|Trace|HttpClient|Res: HTTP/2.0 [GET] https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/kickasstorrents-ws/results/torznab/api?t=search&cat=6000,103510&extended=1&apikey=(removed)&offset=0&limit=100: 200.OK (540 bytes) (191 ms)
2025-03-28 21:45:25.1|Trace|WhisparrErrorPipeline|Handling Exception
2025-03-28 21:45:25.1|Warn|WhisparrErrorPipeline|Invalid request Validation failed: 
 -- : Query successful, but no results in the configured categories were returned from your indexer. This may be an issue with the indexer or your indexer category settings.
2025-03-28 21:45:25.1|Trace|Http|Res: 3977 [POST] /api/v3/indexer/test: 400.BadRequest (207 ms)
2025-03-28 21:45:25.1|Debug|Api|[POST] /api/v3/indexer/test: 400.BadRequest (207 ms)
2025-03-28 21:45:28.6|Trace|Http|Req: 3978 [GET] /logfile/whisparr.trace.txt (from 195.*.*.48 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:136.0) Gecko/20100101 Firefox/136.0)


And here's a trace log for TheRARBG torznab in Jackett which works correctly.

2025-03-28 21:47:17.6|Trace|Http|Req: 3985 [POST] /api/v3/indexer/test (from 195.*.*.48 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:136.0) Gecko/20100101 Firefox/136.0)
2025-03-28 21:47:17.6|Debug|Torznab|Downloading Feed https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/therarbg/results/torznab/api?t=search&cat=6000,6010,6030,6040,6050&extended=1&apikey=(removed)&offset=0&limit=100
2025-03-28 21:47:17.6|Trace|HttpClient|Req: [GET] https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/therarbg/results/torznab/api?t=search&cat=6000,6010,6030,6040,6050&extended=1&apikey=(removed)&offset=0&limit=100
2025-03-28 21:47:17.6|Trace|ConfigService|Using default config value for 'proxyenabled' defaultValue:'False'
2025-03-28 21:47:17.6|Trace|HttpClient|Res: HTTP/2.0 [GET] https://gasbandit-jackett.cloud.seedboxes.cc/api/v2.0/indexers/therarbg/results/torznab/api?t=search&cat=6000,6010,6030,6040,6050&extended=1&apikey=(removed)&offset=0&limit=100: 200.OK (114776 bytes) (30 ms)
2025-03-28 21:47:17.6|Trace|TorznabRssParser|Parsed: (Clipped out about 50 results here -GB)
2025-03-28 21:47:17.6|Trace|Http|Res: 3985 [POST] /api/v3/indexer/test: 200.OK (53 ms)
2025-03-28 21:47:17.6|Debug|Api|[POST] /api/v3/indexer/test: 200.OK (53 ms)
2025-03-28 21:47:20.7|Trace|Http|Req: 3986 [GET] /logfile/whisparr.trace.txt (from 195.*.*.48 Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:136.0) Gecko/20100101 Firefox/136.0)

Here's the Jackett log comparing the two tests

![Image](https://github.com/user-attachments/assets/cd443bce-22d8-4804-93af-c35a6ef7c0a6)

But doing a search from within jackett itself with a blank query does return results

![Image](https://github.com/user-attachments/assets/3665482a-2dcd-45e4-9104-d1659d7fecae)

**Comment by GasBandit** [comment]
Waaaiit a minute... I might be on to something.... if I limit the jackett query to just the same categories Whisparr requests, then it returns no results on a blank query... but if I search for a specific term, it returns results....

Ok... confirmed it.   The key to getting Jackett torznabs that seem recalcitrant to "work" when testing is to not narrow down your categories to the merely appropriate ones.  Have it search ALL the categories (or at least the most popular ones like movies and shows) and it'll pass muster.  Thanks for your time and eyestrain.  This got 1337x and others working again.
